src/ngraph/runtime/cpu/kernel/one_hot.hpp

```diff
@@ -0,0 +1,90 @@
+/*******************************************************************************
+* Copyright 2018 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#pragma once
+
+#include "ngraph/runtime/cpu/cpu_eigen_utils.hpp"
+#include "ngraph/runtime/reference/one_hot.hpp"
+#include "ngraph/shape.hpp"
+
+namespace ngraph
+{
+    namespace runtime
+    {
+        namespace cpu
+        {
+            namespace kernel
+            {
+                template <typename ElementType>
+                void one_hot_rank_0(void* arg,
+                                    void* out,
+                                    const Shape& arg_shape,
+                                    const Shape& out_shape,
+                                    size_t one_hot_axis)
+
+                {
+                    eigen::EigenVector<ElementType> arg_vector(
```

Comment:
DEX shouldn't be using eigen core since that brings no real benefit. Could you switch to Eigen::Tensor?
If this isn't used in current or future models we care about (@jbobba should know this, I think?), then we could go with this